tests/test.wakelock.js

```diff
@@ -0,0 +1,198 @@
+/* jshint expr: true */
+
+import WakeLockManager from '../core/util/wakelock.js';
+
+class FakeWakeLockSentinal extends EventTarget {
+    constructor() {
+        super();
+        this.released = false;
+    }
+
+    async release() {
+        if (this.released) {
+            return;
+        }
+        this.released = true;
+        this.dispatchEvent(new Event("release"));
+    }
+}
+
+function waitForStateTransition(wakelockManager, newState) {
+    const {promise, resolve} = Promise.withResolvers();
+
+    const eventListener = (event) => {
+        console.warn(`Got state change: ${event.oldState} -> ${event.newState}`);
```

Comment:
This should probably be removed as it will generate noise when running the tests.